py/src/braintrust/oai.py

```diff
@@ -746,3 +746,28 @@ def _parse_metrics_from_usage(usage: Dict[str, Any]) -> Dict[str, Any]:
 
 def _is_numeric(v):
     return isinstance(v, (int, float, complex))
+
+
+def prettify_params(params: Dict[str, Any]) -> Dict[str, Any]:
+    ret = {**params}
+    if "response_format" in ret:
+        ret["response_format"] = serialize_response_format(ret["response_format"])
```

Comment:
my understanding here is that the `response_format` used to have a pydantic class/model but that wasn't serializable when that param was logged. 

should we attempt to recurse through the objects passed to the arguments try to serialize automatically? i.e. if we see a BaseModel subclass we could `model_json_schema` as default.

I suppose it may be wasteful and may hide problems until someone notices vs. this way it's performant and directed?

Reply:
We could but I'd rather start with something targeted